lib/garbage_collector.rb

```diff
@@ -0,0 +1,60 @@
+require_relative 'bintray_client'
+require_relative 'github_client'
+
+class GarbageCollector
+
+  def initialize(bintray_user, bintray_key, verbose = false, dry_run = false)
```

Comment:
nit/suggestion: now that Ruby allows it I like to use named arguments whenever possible (especially for the ones with default values as they can be omitted) to make APIs clearer, e.g.:
```suggestion
  def initialize(bintray_user, bintray_key, verbose: false, dry_run: false)
```

Then at call site it would look like:
```
GarbageCollector.new(bintray_user, bintray_key, verbose: verbose, dry_run: dry_run).run(projects)
```

But it will also allow us to pass `dry_run: true` without having to explicitly pass a value (and thus using the default) for `verbose` (which wouldn't be possible without named args)

nbd tho, just thought I'd mention it in case you didn't know about this feature but feel free to keep as is if you prefer.

Reply:
Thanks for bringing this up. I vaguely knew about this feature, but never used it. It makes total sense and I love how it makes up (partly) the lack of compiler to help know what parameter to pass where.

> But it will also allow us to pass dry_run: true without having to explicitly pass a value (and thus using the default) for verbose (which wouldn't be possible without named args)

This is even more so a good reason to use it. Although, in the particular case of how this object is used in the `bin/` script, I think we'll always want to explicitly pass both values. I would have liked to make them options (`--dry-run`, `--verbose`), but run out of time. I might do that in a followup PR when I'll add more projects to the list.